expyfun/stimuli/_tracker.py

```diff
@@ -738,19 +770,43 @@ def __init__(self, trackers, max_lag=1, rand=None):
         self._response_history = np.array([], dtype=int)
         self._x_history = np.array([], dtype=float)
 
-    def __getitem__(self, key):
-        return self._trackers[key]
+        self._dealer_id = id(self)
+        self._callback('dealer_identity', json.dumps(dict(
+            dealer_id=self._dealer_id)))
+
+        self._callback('dealer_%i_init' % self._dealer_id, json.dumps(dict(
+            trackers=[s._tracker_id for s in self._trackers.ravel()],
+            shape=self._shape,
+            max_lag=self._max_lag,
+            pace_rule=self._pace_rule)))
 
     def __iter__(self):
-        self._index = 0
         return self
 
     def next(self):
-        if self._index == self._trackers.size:
+        """Selects the tracker from which the next trial should be run
+
+        Returns
+        -------
+        subscripts : list-like
+            The position of the selected tracker.
+        x_current : float
+            The level of the selected tracker.
+        """
+        if self.stopped:
             raise(StopIteration)
-        t = self._trackers.flat[self._index]
-        self._index += 1
-        return t
+        if not self._trial_complete:
+            # Chose a new tracker before responding, so record non-response
+            self._response_history = np.append(self._response_history,
+                                               np.nan)
+        self._trial_complete = False
+        self._current_tracker = self._pick()
+        self._tracker_history = np.append(self._tracker_history,
+                                          self._current_tracker)
+        ss = np.unravel_index(self._current_tracker, self.shape)
+        level = self._trackers.flat[self._current_tracker].x_current
```

Comment:
Okay if that's the case update the docs to reflect this -- currently they say it's useful in a way that makes it sound like TrackerDealer's API needs it (which the user does not care about), but it's (also) useful so that the level gets documented properly (which the user will care about).